### PR TITLE
Implement order creation endpoint

### PR DESCRIPTION
**Pull request by kq-li**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
     "reflect-metadata": "^0.1.10",
     "request": "^2.88.0",
     "square-connect": "^2.20190213.0",
+    "square-connect-ts": "^1.1.0",
```
Comment (sahamanish99) [inline package.json]: do we need both square-connect and square-connect-ts?

File: src/common/constants.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const USD_IDENTIFIER = 'USD';
```
Comment (sahamanish99) [inline src/common/constants.ts]: can we just call this USD instead of USD_IDENTIFIER, or is there a naming convention for currencies?

File: src/routers/CreateOrderRouter.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import TransactionsRepo from '../repos/TransactionsRepo';
+import UsersRepo from '../repos/UsersRepo';
+
+class RegisterRouter extends ApplicationRouter<SerializedTransaction> {
+  constructor() {
+    super('POST');
+  }
+
+  getPath(): string {
+    return '/register/';
```
Comment (sahamanish99) [inline src/routers/CreateOrderRouter.ts]: Should change this to `/placeOrder/` or something appropriate

File: src/routers/CreateOrderRouter.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      user.customerId,
+      cardNonce,
+      squareOrder.order.id,
+      squareOrder.order.totalMoney.amount,
+    );
+    return transaction.serialize();
+  }
+}
+
+export default new RegisterRouter().router;
```
Comment (sahamanish99) [inline src/routers/CreateOrderRouter.ts]: change this too

File: src/Api.ts [hunk 5]
```diff
@@ -1,13 +1,13 @@
 import bodyParser from 'body-parser';
 import { Router } from 'express';
-import ApplicationAPI from './appdev/ApplicationAPI';
+import ApplicationApi from './appdev/ApplicationApi';
```
Comment (sahamanish99) [inline src/Api.ts]: why the change from API to Api?

File: src/routers/CreateOrderRouter.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  getPath(): string {
+    return '/register/';
+  }
+
+  async content(req: Request): Promise<SerializedTransaction> {
+    const { cardNonce, itemVariations, taxes } = req.body;
+    const sessionToken = utils.extractAuthToken(req.get('Authorization'));
+    const user = await UsersRepo.getUserBySessionToken(sessionToken);
```
Comment (sahamanish99) [inline src/routers/CreateOrderRouter.ts]: shouldn't we call `.serialize()` to get the SerializedUser?

File: src/Api.ts (hunk 5 above)
Comment (meganle) [reply src/Api.ts]: yeah should be API

File: src/common/SquareApi.ts [hunk 7]
```diff
@@ -0,0 +1,121 @@
+import squareConnect from 'square-connect';
```
Comment (meganle) [inline src/common/SquareApi.ts]: File should be `SquareAPI`

File: src/repos/TransactionsRepo.ts [hunk 8]
```diff
@@ -1,32 +1,43 @@
 import { getConnectionManager, Repository } from 'typeorm';
 import Transaction from '../entities/Transaction';
```
Comment (meganle) [inline src/repos/TransactionsRepo.ts]: the newline should be between line 1 and 2

File: src/routers/CreateOrderRouter.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import { Request } from 'express';
+
+import ApplicationRouter from '../appdev/ApplicationRouter';
+import SquareApi from '../common/SquareApi';
+import { SerializedTransaction } from '../common/types';
+import utils from '../common/utils';
+import TransactionsRepo from '../repos/TransactionsRepo';
+import UsersRepo from '../repos/UsersRepo';
+
+class RegisterRouter extends ApplicationRouter<SerializedTransaction> {
```
Comment (meganle) [inline src/routers/CreateOrderRouter.ts]: Change Router name

File: src/Api.ts (hunk 5 above)
Comment (keivanshahida) [reply src/Api.ts]: Agreed ^

File: src/common/SquareApi.ts (hunk 7 above)
Comment (keivanshahida) [reply src/common/SquareApi.ts]: RT

File: src/routers/CreateOrderRouter.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
+import { Request } from 'express';
+
+import ApplicationRouter from '../appdev/ApplicationRouter';
+import SquareApi from '../common/SquareApi';
+import { SerializedTransaction } from '../common/types';
+import utils from '../common/utils';
+import TransactionsRepo from '../repos/TransactionsRepo';
```
Comment (keivanshahida) [inline src/routers/CreateOrderRouter.ts]: Alphabetize please

File: package.json (hunk 1 above)
Comment (kq-li) [reply package.json]: the latter provides typescript bindings ya

File: src/common/constants.ts (hunk 2 above)
Comment (kq-li) [reply src/common/constants.ts]: generally slighty longer and more descriptive constant names are better imo

File: src/Api.ts (hunk 5 above)
Comment (kq-li) [reply src/Api.ts]: see slack message

File: src/routers/CreateOrderRouter.ts (hunk 6 above)
Comment (kq-li) [reply src/routers/CreateOrderRouter.ts]: we don't need to serialize until return

File: src/@types/square-connect.d.ts [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+    order_id: string;
+    tenders?: Tender[];
+  };
+  
+  class CreateCustomerRequest extends Request {
+    given_name: string;
+    family_name: string;
+    email_address: string;
+    phone_number: string;
+  }
```
Comment (meganle) [inline src/@types/square-connect.d.ts]: any reason why these classes below don't have a semicolon at the end?

Comment (kq-li) [reply src/@types/square-connect.d.ts]: classes don't end in semicolons according to tslint

Comment (meganle) [reply src/@types/square-connect.d.ts]: some of the classes above have semicolons

Comment (kq-li) [reply src/@types/square-connect.d.ts]: ah whoops good catch
